src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/SharedAccessSignatureCreator.cs

```
@@ -0,0 +1,116 @@
+            {
+                try
+                {
+                    azureCredential.Create(identity, secretString);
+                }
+                catch (Exception ex)
+                {
+                    throw new FailedOperationException("Create credential failed.", ex);
+                }
+            }
+            return azureCredential;
+        }
+
+        /// <summary>
+        /// Create Shared Access Policy for container 
+        /// Default Accesss permission is Write/List/Read/Delete
+        /// </summary>
+        /// <param name="container"></param>
+        /// <param name="policyName"></param>
+        /// <param name="selectedSaredAccessExpiryTime"></param>
+        public Uri GetServiceSasUriForContainer(BlobContainerClient containerClient,
+                                          string storedPolicyName = null)
+        {
+            // Check whether this BlobContainerClient object has been authorized with Shared Key.
+            if (containerClient.CanGenerateSasUri)
+            {
+                // Create a SAS token that's valid for one year.
+                BlobSasBuilder sasBuilder = new BlobSasBuilder()
+                {
+                    BlobContainerName = containerClient.Name,
+                    Resource = "c"
+                };
+
+                if (storedPolicyName == null)
+                {
+                    sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddYears(1);
+                    sasBuilder.SetPermissions(BlobContainerSasPermissions.Read | BlobContainerSasPermissions.List | BlobContainerSasPermissions.Write | BlobContainerSasPermissions.Delete);
+                }
+                else
+                {
+                    sasBuilder.Identifier = storedPolicyName;
+                }
+                Uri sasUri = containerClient.GenerateSasUri(sasBuilder);
+
+                return sasUri;
+            }
+            else
+            {
+                return null;
```

The return type should be `Uri?` if it can return null, but wouldn't you want to throw an exception if it can't generate the SAS URI?

---

Replaced `return null` with an exception throw